From de53ebc6f81a8080ce9f135f53044dfbc8b5faee Mon Sep 17 00:00:00 2001
From: Paul Spooren <mail@aparcar.org>
Date: Thu, 9 Dec 2021 06:14:27 -1000
Subject: [PATCH 2/4] package: store scripts in tmp

Signed-off-by: Paul Spooren <mail@aparcar.org>
---
 src/package.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/package.c b/src/package.c
index 5aee917..2a4d80e 100644
--- a/src/package.c
+++ b/src/package.c
@@ -799,7 +799,7 @@ void apk_ipkg_run_script(struct apk_installed_package *ipkg,
 	argv[0] = (char *) apk_script_types[type];
 
 	/* Avoid /tmp as it can be mounted noexec */
-	snprintf(fn, sizeof(fn), "var/cache/misc/" PKG_VER_FMT ".%s",
+	snprintf(fn, sizeof(fn), "tmp/cache/misc/" PKG_VER_FMT ".%s",
 		PKG_VER_PRINTF(pkg),
 		apk_script_types[type]);
 
@@ -811,10 +811,10 @@ void apk_ipkg_run_script(struct apk_installed_package *ipkg,
 		if (faccessat(db->root_fd, "tmp", F_OK, 0) != 0)
 			mkdirat(db->root_fd, "tmp", 01777);
 		make_device_tree(db);
-		if (faccessat(db->root_fd, "var/cache/misc", F_OK, 0) != 0) {
-			mkdirat(root_fd, "var", 0755);
-			mkdirat(root_fd, "var/cache", 0755);
-			mkdirat(root_fd, "var/cache/misc", 0755);
+		if (faccessat(db->root_fd, "tmp/cache/misc", F_OK, 0) != 0) {
+			mkdirat(root_fd, "tmp", 0755);
+			mkdirat(root_fd, "tmp/cache", 0755);
+			mkdirat(root_fd, "tmp/cache/misc", 0755);
 		}
 	}
 
-- 
2.35.1

