From 5dd7a14013dbce2bad0368cdb321810e388773f3 Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Fri, 10 Dec 2021 14:04:21 -0600
Subject: [PATCH 08/16] libfetch: use static_deps when compiling libfetch under
 meson

---
 libfetch/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libfetch/meson.build b/libfetch/meson.build
index 90e133b..18124f7 100644
--- a/libfetch/meson.build
+++ b/libfetch/meson.build
@@ -38,6 +38,7 @@ libfetch = static_library(
 	'fetch',
 	libfetch_src,
 	c_args: libfetch_cargs,
+	dependencies: static_deps,
 )
 
 libfetch_dep = declare_dependency(
-- 
2.34.1

